presto-hive-metastore/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java

```diff
@@ -34,16 +36,42 @@
     private final List<String> bucketedBy;
     private final int bucketCount;
     private final List<SortingColumn> sortedBy;
+    private final Optional<List<Type>> types;
+    private final BucketingType bucketingType;
+
+    public HiveBucketProperty(
```

Comment:
How strongly do we feel about keeping this constructor? How many places have to be refactored if we are to remove it?

Reply:
More than 10 call sites,   since "Optional<List<Type>> types" is optional, keeping the old constructor has some benefits.

Reply:
Done